### PR TITLE
Add the option to avoid loading the checkpoint twice

### PR DESCRIPTION
**Pull request by xuefgu**
# Description

Add the option to avoid loading the checkpoint twice.

It's protected with the `load_checkpoint_only_once` flag.

# Tests

Manually tested.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: src/maxtext/configs/types.py [hunk 1]
```diff
@@ -288,6 +288,7 @@ class Checkpointing(BaseModel):
   lora_input_adapters_path: PathStr = Field("", description="Input GCS path for LoRA adapters.")
   load_full_state_path: PathStr = Field("", description="Loads the complete training state from a checkpoint path.")
   enable_checkpointing: bool = Field(True, description="If True, enables saving checkpoints during training.")
+  load_checkpoint_only_once: bool = Field(False, description="If True, deep copy the reference model to the actor model.")
```
Comment (bvandermoon) [inline src/maxtext/configs/types.py:291]: What was the reason for loading this twice previously?

Comment (xuefgu) [reply src/maxtext/configs/types.py:291]: The reference model has to stay the same while the actor model is being updated during training. They can't share the same parameters.

File: src/MaxText/rl/train_rl.py [hunk 2]
```diff
@@ -530,11 +538,23 @@ def rl_train(trainer_config, sampler_config, trainer_devices, sampler_devices):
...
+    max_logging.log("Capturing reference model state before training.")
+    ref_state_before = nnx.to_pure_dict(nnx.state(reference_model.base, nnx.Param))
+
   max_logging.warning("Starting RL training...")
 
   with reference_mesh, nn_partitioning.axis_rules(trainer_config.logical_axis_rules):
     rl_trainer.train(train_dataset)
 
+  if trainer_config.load_checkpoint_only_once:
+    max_logging.log("Checking if reference model state changed during training.")
```
Comment (khatwanimohit) [inline src/MaxText/rl/train_rl.py:551]: can you share how much time it takes to do this check for our Qwen model. This seems to be an expensive check to do, if it is then can we hide it behind a flag

Comment (xuefgu) [reply src/MaxText/rl/train_rl.py:551]: Llama70B takes ~6 secs to check in my manual testing. I would let folks routinely running Qwen provide the numbers for it :-)

That said, the whole block is a temporary safety check before the `load_checkpoint_only_once` flag is removed. A temporary 20 sec delay for a 235B model may not be worth adding complexities/mental load that comes with another flag.

Comment (khatwanimohit) [reply src/MaxText/rl/train_rl.py:551]: agreed, thanks for the number for Llama70B, its pretty cheap.

**Review by khatwanimohit — APPROVED** [review]
